src/conventional/amv_ssec_ascii2ioda.py

```diff
@@ -0,0 +1,320 @@
+        sys.exit()
+
+    # prepare global attributes we want to output in the file,
+    # in addition to the ones already loaded in from the input file
+    GlobalAttrs = {
+        'sourceFiles': ", ".join(file_names),
+        'datetimeReference': datetimeRef
+    }
+
+    nlocs = len(data['dateTime'])
+    logging.info(f" found a total of {nlocs} observations")
+    DimDict = {'nlocs': nlocs}
+
+    varDims = {}
+    for key in varDict.keys():
+        variable = varDict[key][0]
+        varDims[variable] = ['nlocs']
+
+    varAttrs = DefaultOrderedDict(lambda: DefaultOrderedDict(dict))
+
+    # Set units of the MetaData variables and all _FillValues.
+    for key in meta_keys:
+        dtypestr = locationKeyList[meta_keys.index(key)][1]
+        if locationKeyList[meta_keys.index(key)][2]:
+            varAttrs[(key, metaDataName)]['units'] = locationKeyList[meta_keys.index(key)][2]
+        varAttrs[(key, metaDataName)]['_FillValue'] = missing_vals[dtypestr]
+
+    # Set units and FillValue attributes for groups associated with observed variable.
+    for key in varDict.keys():
+        variable = varDict[key][0]
+        units = varDict[key][1]
+        varAttrs[(variable, obsValName)]['units'] = units
+        varAttrs[(variable, obsErrName)]['units'] = units
+        varAttrs[(variable, obsValName)]['coordinates'] = 'longitude latitude'
+        varAttrs[(variable, obsErrName)]['coordinates'] = 'longitude latitude'
+        varAttrs[(variable, qcName)]['coordinates'] = 'longitude latitude'
+        varAttrs[(variable, obsValName)]['_FillValue'] = float_missing_value
+        varAttrs[(variable, obsErrName)]['_FillValue'] = float_missing_value
+        varAttrs[(variable, qcName)]['_FillValue'] = int_missing_value
+
+    # Fill the final IODA data:  MetaData then ObsValues, ObsErrors, and QC
+    for key in meta_keys:
+        dtypestr = locationKeyList[meta_keys.index(key)][1]
+        ioda_data[(key, metaDataName)] = np.array(data[key], dtype=dtypes[dtypestr])
+
+    for key in varDict.keys():
+        variable = varDict[key][0]
+        logging.info(f" the variable: {variable} will be placed into ioda_data")
+        ioda_data[(variable, obsValName)] = np.array(data[variable], dtype=np.float32)
+        ioda_data[(variable, obsErrName)] = np.full(nlocs, 3.0, dtype=np.float32)
+        ioda_data[(variable, qcName)] = np.full(nlocs, 2, dtype=np.int32)
+
+    logging.debug("Writing file: " + output_file)
+
+    # setup the IODA writer and write everything out.
+    writer = iconv.IodaWriter(output_file, locationKeyList, DimDict)
+    writer.BuildIoda(ioda_data, varDims, varAttrs, GlobalAttrs)
+
+    logging.info("--- {:9.4f} total seconds ---".format(time.time() - start_time))
+
+
+def read_file(file_name, data):
+
+    with open(file_name, newline='') as f:
+        reader = csv.DictReader(f, skipinitialspace=True, delimiter=' ')
+        keyerr = False
+        freqerr = False
+        saterr = False
```

Comment:
It looks like the purpose of `freqerr` and `saterr` is to print the warning message once (the first time) when encountering these errors (ie, so you don't get thousands of warning messages). Is this correct? If so, one issue I can think of is that if you encounter more than one unknown channel/satellite type you only get warned about the first one that you encounter. To address that issue you could instead create top-level dictionaries for unknown channels and another for unknown satellites. Then in the `get_frequency` and `get_id` functions, when an unknown type is encountered do something like: `unknown_chan[obs_type] = 1`. It doesn't matter what the value you assign is, the idea is that you get one key for each unknown channel/satellite id and you can then print out all the unknown ids at the end of the script.

Reply:
Thanks, the change has been made